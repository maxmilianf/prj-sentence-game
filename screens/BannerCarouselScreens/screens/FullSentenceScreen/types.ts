import { TextStyle, ViewStyle } from 'react-native';

export type FullSentenceScreenStyles = {
  container: ViewStyle;
  sentence: TextStyle;
  takeBackButton: ViewStyle;
};
