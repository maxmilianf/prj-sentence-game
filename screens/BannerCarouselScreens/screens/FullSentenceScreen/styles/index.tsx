import { StyleSheet } from 'react-native';
import { FullSentenceScreenStylesProps } from '../types';

export const fullSentenceScreenStyles =
  StyleSheet.create<FullSentenceScreenStylesProps>({
    container: {
      flex: 1,
      alignItems: 'center',
      marginTop: 150,
    },
    sentence: {
      opacity: 0.8,
      marginTop: 20,
      width: '75%',
      textAlign: 'center',
      fontStyle: 'italic',
    },
    takeBackButton: {
      backgroundColor: 'black',
      opacity: 0.8,
      marginTop: 350,
    },
  });
