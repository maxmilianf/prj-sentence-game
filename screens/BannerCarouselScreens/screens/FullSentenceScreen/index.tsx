import React, { useContext } from 'react';
import { View } from 'react-native';
import { Headline, Subheading, Button } from 'react-native-paper';
import { fullSentenceScreenStyles } from './styles';
import { SentenceContext } from '../../../../context/SentenceContext';

interface fullSentenceScreenProps {
  slideCarouselToStart: () => void;
}

const FullSentenceScreen = ({
  slideCarouselToStart = () => {},
}: fullSentenceScreenProps) => {
  const { fullSentence, displaySentence } = useContext(SentenceContext);
  const errorMsg = 'You did not finish your sentence';
  const displayFullSentence = displaySentence ? fullSentence : errorMsg;
  const styles = fullSentenceScreenStyles;

  return (
    <View style={styles.container}>
      <Headline style={{ opacity: 0.8 }}>Your sentence</Headline>
      <Subheading style={styles.sentence}>{displayFullSentence}</Subheading>
      {/* 
        // @ts-ignore react-native-paper version most likely*/}
      <Button
        onPress={slideCarouselToStart}
        style={styles.takeBackButton}
        mode='contained'
      >
        Create new sentence
      </Button>
    </View>
  );
};

export default FullSentenceScreen;
