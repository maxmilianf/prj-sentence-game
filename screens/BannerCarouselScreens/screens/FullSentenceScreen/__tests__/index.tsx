import React from 'react';
import renderer from 'react-test-renderer';

import FullSentenceScreen from '../index';

it('FullSentenceScreen should render correctly', () => {
  const tree = renderer
    .create(
      <FullSentenceScreen slideCarouselToStart={() => {}}></FullSentenceScreen>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
