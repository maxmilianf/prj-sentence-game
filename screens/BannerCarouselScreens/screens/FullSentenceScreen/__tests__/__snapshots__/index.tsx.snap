// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FullSentenceScreen should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "marginTop": 150,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "System",
          "fontWeight": "400",
        },
        Object {
          "textAlign": "left",
        },
        Array [
          Object {
            "textAlign": "left",
          },
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "fontFamily": "System",
            "fontWeight": "400",
            "writingDirection": "ltr",
          },
          Array [
            Object {
              "fontSize": 24,
              "letterSpacing": 0,
              "lineHeight": 32,
              "marginVertical": 2,
            },
            Object {
              "opacity": 0.8,
            },
          ],
        ],
      ]
    }
  >
    Your sentence
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "System",
          "fontWeight": "400",
        },
        Object {
          "textAlign": "left",
        },
        Array [
          Object {
            "textAlign": "left",
          },
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "fontFamily": "System",
            "fontWeight": "400",
            "writingDirection": "ltr",
          },
          Array [
            Object {
              "fontSize": 16,
              "letterSpacing": 0.5,
              "lineHeight": 24,
              "marginVertical": 2,
            },
            Object {
              "fontStyle": "italic",
              "marginTop": 20,
              "opacity": 0.8,
              "textAlign": "center",
              "width": "75%",
            },
          ],
        ],
      ]
    }
  >
    
  </Text>
  <View
    collapsable={false}
    style={
      Object {
        "backgroundColor": "black",
        "borderColor": "transparent",
        "borderRadius": 4,
        "borderStyle": "solid",
        "borderWidth": 0,
        "elevation": 2,
        "marginTop": 350,
        "minWidth": 64,
        "opacity": 0.8,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 0.75,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 1.5,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 4,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <Text
          numberOfLines={1}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "textAlign": "left",
              },
              Array [
                Object {
                  "letterSpacing": 1,
                  "marginHorizontal": 16,
                  "marginVertical": 9,
                  "textAlign": "center",
                },
                undefined,
                Object {
                  "textTransform": "uppercase",
                },
                Object {
                  "color": "#ffffff",
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                Object {
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                undefined,
              ],
            ]
          }
        >
          Create new sentence
        </Text>
      </View>
    </View>
  </View>
</View>
`;
